docs/conandata_yml_format.md

```diff
@@ -0,0 +1,228 @@
+
+```
+sources:
+  "4.5.0":
+    - sha256: dde4bf8d6639a5d3fe34d5515eab4a15669ded609a1d622350c7ff20dace1907
+      url: https://github.com/opencv/opencv/archive/4.5.0.tar.gz
+    - sha256: a65f1f0b98b2c720abbf122c502044d11f427a43212d85d8d2402d7a6339edda
+      url: https://github.com/opencv/opencv_contrib/archive/4.5.0.tar.gz
+```
+
+#### Different source code archives per configuration
+
+This is the most advanced and sophisticated use-case, but no so common. Some projects may provide different sources for different platforms for awkward reasons, it could be expressed as:
+
+```
+sources:
+  "0066":
+    "Macos":
+      "apple-clang":
+        "x86":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_32bit/packages/cspice.tar.Z"
+            sha256: "9a4b5f674ea76821c43aa9140829da4091de646ef3ce40fd5be1d09d7c37b6b3"
+        "x86_64":
+          - url: "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N0066/C/MacIntel_OSX_AppleC_64bit/packages/cspice.tar.Z"
+            sha256: "f5d48c4b0d558c5d71e8bf6fcdf135b0943210c1ff91f8191dfc447419a6b12e"
+```
+
+This approach requires a special code within [build](https://docs.conan.io/en/latest/reference/conanfile/methods.html#build) method to handle.
+
+## patches
+
+Sometimes sources provided by project require patching for various reasons. The `conandata.yml` file is the right place to indicate this information as well.
+
+This section follows the same pattern as the `sources` above: one entry per version with a list of patches to apply.
+
+```yaml
+patches:
+  "1.2.0":
+    - patch_file: "patches/1.2.0-002-link-core-with-find-library.patch"
+      base_path: "source_subfolder"
+      url: "https://a-url-to-a-pull-request-mail-list-topic-issue-or-question"
+      sha256: "qafe4rq54533qa43esdaq53ewqa5"
+      patch_type: "portability"
+      patch_source: ...
+      patch_description: "Link CoreFoundation and CoreServices with find_library"
+```
+
+### Patches fields
+
+#### patch_file
+
+Patch file might be committed to the conan-center-index, near to the conanfile (usually, into the `patches` sub-directory). Such patch files usually have either `.diff` or `.patch` extension.
+The recommended way to generate such patches is [git format-patch](https://git-scm.com/docs/git-format-patch). The path to the patch is relative to the directory containing `conandata.yml` and `conanfile.py`.
+
+#### base_path
+
+Specifies a sub-directory in project's sources to apply patch. This directory is relative to the [source_folder](https://docs.conan.io/en/latest/reference/conanfile/attributes.html?highlight=source_folder#source-folder). Usually, it would be a `source_subfolder`, but could be a lower-level sub-directory (e.g. if it's a patch for a submodule).
+
+#### url
+
+The patch could be taken from the existing resource (e.g. some patches are distributed alongside the source tarball). See `url` section of the `sources`.
```

Comment:
Trying to capture https://github.com/conan-io/conan-center-index/pull/11713/files#r923287561 since I found it a little unclear

```suggestion
When the patch is taken from an existing resource (e.g. some patches are distributed alongside the source tarball) this should point to the same `<filename>.<extension>` as used in the `patch_file`. [`patch_source`](#patch-source) may point to the same resource or to a more human-friendly view. See `url` section of the `sources` for details.
```